src/vue/common/CollectionLoader.spec.js

```diff
@@ -0,0 +1,209 @@
+    const spy = sinon.stub().resolves({
+      fetchNext,
+      data: mockData
+    })
+
+    await wrapper.vm.loadPage('some-event', spy)
+
+    expect(wrapper.vm.isCollectionFetched).to.be.false
+  })
+
+  it('isCollectionFetched true when data is fully loaded', async () => {
+    const fetchNext = () => {}
+    const mockData = []
+    const pageLimit = 10
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader: sinon.stub().resolves({
+          data: mockData,
+          fetchNext
+        }),
+        pageLimit
+      },
+      localVue
+    })
+    mockData.length = pageLimit - 1
+    const spy = sinon.stub().resolves({
+      fetchNext,
+      data: mockData
+    })
+
+    await wrapper.vm.loadPage('some-event', spy)
+
+    expect(wrapper.vm.isCollectionFetched).to.be.true
+  })
+
+  it('loadPage delegates error to ErrorHandler when got one', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const error = new Error('Error')
+    const spy = sinon.stub().throws(error)
+    const spyError = sinon.stub(ErrorHandler, 'process')
+
+    await wrapper.vm.loadPage('some-event', spy)
+
+    expect(spyError.withArgs(error).calledOnce).to.be.true
+  })
+
+  it('loadFirstPage calls loadPage', () => {
+    const firstPageLoader = sinon.stub().resolves({
+      data: [],
+      fetchNext: () => {}
+    })
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+
+    const spy = sinon.stub(wrapper.vm, 'loadPage')
+
+    wrapper.vm.loadFirstPage()
+
+    expect(spy.withArgs('first-page-load', firstPageLoader).calledOnce)
+      .to.be.true
+  })
+
+  it('loadNextPage calls loadPage', () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader: () => { return { data: [] } }
+      },
+      localVue
+    })
+
+    const spy = sinon.stub(wrapper.vm, 'loadPage')
+    const nextPageLoader = sinon.stub().resolves({})
+
+    wrapper.setData({
+      nextPageLoader
+    })
+
+    wrapper.vm.loadNextPage('next-page-load', nextPageLoader)
+
+    expect(spy.withArgs('next-page-load', nextPageLoader).calledOnce)
+      .to.be.true
+  })
+
+  it('button is displayed when not all data is loaded', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      }
+    })
+    wrapper.setData({ isCollectionFetched: false })
+
+    expect(wrapper.contains('button')).to.be.true
+  })
+
+  it('button is not displayed when all data is loaded', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      }
+    })
+    wrapper.setData({ isCollectionFetched: true })
+
+    expect(wrapper.contains('button')).to.be.false
+  })
+  it('calls firstPageLoader when creating a component', () => {
```

Comment:
firstPageLoader is immediately called after component created

Reply:
ok